tests not compatible with gradle

The current tests cannot run with Gradle, this is because:

- Gradle starts build with a separate JVM
- in the TestSever, after test, the server will shutdown, which calls System.exit()
- System.exit() will quit the JVM and then Gradle cannot connect to the JVM to run other tests
- this is a known issue with Gradle when using System.exit()
- this error won't happen when running tests in parallel due to the sever won't shut down if other tests pending, but somehow this doesn't apply to the gradle test
 
```
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Unexpected exception thrown.
org.gradle.internal.remote.internal.MessageIOException: Could not write '/127.0.0.1:53061'.
        at org.gradle.internal.remote.internal.inet.SocketConnection.flush(SocketConnection.java:140)
        at org.gradle.internal.remote.internal.hub.MessageHub$ConnectionDispatch.run(MessageHub.java:333)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: An existing connection was forcibly closed by the remote host
        at java.base/sun.nio.ch.SocketDispatcher.write0(Native Method)
        at java.base/sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
        at java.base/sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:113)
        at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:58)
        at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:50)
        at java.base/sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:466)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.writeWithNonBlockingRetry(SocketConnection.java:279)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.writeBufferToChannel(SocketConnection.java:267)
        at org.gradle.internal.remote.internal.inet.SocketConnection$SocketOutputStream.flush(SocketConnection.java:261)
        at org.gradle.internal.remote.internal.inet.SocketConnection.flush(SocketConnection.java:138)
        ... 7 more
```

--- a/light-rest-4j/src/main/resources/templates/rest/testServer.rocker.raw
+++ b/light-rest-4j/src/main/resources/templates/rest/testServer.rocker.raw
@@ -40,10 +40,10 @@ public class TestServer extends ExternalResource {
         finally {
             refCount.getAndIncrement();
         }
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> clean()));
     }
 
-    @@Override
-    protected void after() {
+    protected void clean() {
         refCount.getAndDecrement();
         if (refCount.get() == 0) {
             Server.stop();

